students/csdotson/lesson04/mailroom_2.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+# Lesson 4 - Mailroom... part 2
+
+### Function Definitions ###
+def prompt(prompt, menu):
+    # Create menu based on arguments
+    while True:
+        response = input(prompt)
+        if menu[response]() == "quit":
+            break
+
+
+def send_thank_you():
+    # Create sub_prompt for creating thank you's and adding new donors/donations
+    prompt(thank_you_prompt, thank_you_menu)
+
+
+def add_donation():
+    # Append donation to existing donor, or add donor/donation to existing list
+    response = input("\nPlease enter a donor name: ")
+    if response not in donations.keys():
+        new_donor = response
+        new_donation = input(f"Please enter a donation amount for {new_donor}: ")
+        donations[new_donor] = [float(new_donation)]
+        print_email(new_donor, new_donation)
+    else:
+         current_donor = response
+         new_donation = input(f"Please enter a donation amount for {current_donor}: ")
+         donations[current_donor].append(float(new_donation))
+         print_email(current_donor, new_donation)
+
+
+def write_letters():
+    # For all donors, generates a thank you letter, and writes it to disk as a text file
+    for donor in donations:
+        file_name = '_'.join(donor.split()) + ".txt"
+        with open(file_name, 'w') as f:
+            f.write(f"Dear {donor},\n\nThank you very much for your generosity! Your most recent gift of ${(donations[donor][-1])} will be put to great use. So far, you've donated a total of ${sum(donations[donor])}!\n\nSincerely,\nThe Team")
```

Comment:
You can use triple quotes `"""` to keep long strings under Pep8 line character limit and they also let you preserve white space in your strings, so this could be written like this:

```
f.write(f"""Dear {donor},
            
            Thank you very much for your generosity! Your most recent gift of ${(donations[donor][-1])} will be 
            put to great use. So far, you've donated a total of ${sum(donations[donor])}!
                    
            Sincerely,
            The Team""")
```